File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        int answer = 0;
        int shot = -1;
        
        Arrays.sort(targets, (o1, o2) -> o1[1] - o2[1]);
        
        for(int[] target: targets){
            int left  = target[0];
            int right = target[1];
            if(left >= shot){
                answer++;
                shot = right;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.kt
import java.util.*;

class Solution {
    fun solution(targets: Array<IntArray>): Int {
        var answer: Int = 0
        var shot = -1
        
        Arrays.sort(targets){ o1, o2 -> o1[1] - o2[1] }
        
        targets.forEach{
            val left = it[0];
            val right = it[1];
            if(left >= shot){
                answer++;
                shot = right
            }
        }
        
        return answer
    }
}